superset/viz.py

```diff
@@ -563,6 +567,46 @@ class TableViz(BaseViz):
     is_timeseries = False
     enforce_numerical_metrics = False
 
+    def process_metrics(self) -> None:
+        """Process form data and store parsed column configs.
+           1. Determine query mode based on form_data params.
+                - Use `query_mode` if it has a valid value
+                - Set as RAW mode if `all_columns` is set
+                - Otherwise defaults to AGG mode
+           2. Determine output columns based on query mode.
+        """
+        super().process_metrics()
```

Comment:
Refactor how data fields are processed for `TableViz`. In general, for all visualizations, we should process all inputs as soon as possible, store the parsed params somewhere, and refer the single source of truth in all future operations.